### PR TITLE
Feature/complete physic class

### PR DESCRIPTION
**Pull request by Hamilcare**
-Created Utils folder
-Move all physics from boat Class to physics Class
-Plug necessary controllers in order to be able to move the boat using keypad
-Turnspeed related to moving speed
Closes : #5 
Closes : #6

### CODE REVIEW

File: TarponSimulator2017/Controller/SceneInGame.cs [hunk 1]
```diff
@@ -13,13 +17,19 @@ public sealed class SceneInGame : Scene
...
 			//this.actions.Add(Keys.Up, new CommandAccelerate(world.playerBoat));
 			//this.actions.Add(Keys.Right, new CommandTurn (world.playerBoat));
 			//this.actions.Add(Keys.Left, new CommandTurn (world.playerBoat));
 		}
 
-
+		public SceneInGame(World world){
+			actions = new Dictionary<Keys,Command> ();
+			Scene.world = world;
+			this.actions.Add (Keys.Up, new CommandAccelerate (world.playerBoat));
```
Comment (superboum) [inline TarponSimulator2017/Controller/SceneInGame.cs]: Well, we'll have to find a linter / format tool.
As here you have a space between Add and parenthesis but not in the 2 following lines.

File: TarponSimulator2017/Core/Boat.cs [hunk 2]
```diff
@@ -5,94 +5,28 @@ namespace Tarpon.Core
...
 
-		public const float FrictionForce = 0.0010f;
-		public const float AccelerationForce = 0.0005f;
-		public const float TurnSpeed = 0.0001f;
-		public const float MaxTurnAngle = 0.02f;
-
-		int elapsedTime; // in milliseconds
-
-		public Boat()
+		public Boat(int AbscisseDepart, int OrdonneeDepart) : base (FrictionForceBoat, AccelerationForceBoat, TurnSpeedBoat, MaxTurnAngleBoat, AbscisseDepart, OrdonneeDepart)
```
Comment (superboum) [inline TarponSimulator2017/Core/Boat.cs]: You put some words in french.
We did not discuss this point, and I don't have strong opinions about it, but we should agree on something, preferably using only one language.

File: TarponSimulator2017/Core/Physics.cs [hunk 3]
```diff
@@ -1,11 +1,137 @@
...
+		
+
+		public float FrictionForce{ get; private set; }
+		public float AccelerationForce{ get; private set; }
+		public float TurnSpeed{ get; private set; }
+		public float MaxTurnAngle{ get; private set; }
+
+		public Vector2 Orientation { get; private set; }
+
+		/** Est recalculé à chaque tour **/
```
Comment (superboum) [inline TarponSimulator2017/Core/Physics.cs]: Same thing here for comments :)

File: TarponSimulator2017/Core/Physics.cs [hunk 4]
```diff
@@ -1,11 +1,137 @@
...
+
+		public void ComputeMovement(int now)
+		{
+			Console.WriteLine ("ComputeMovement triggered");	
+			elapsedTime = now;
+
+			// Friction
+			Acceleration += -Speed * FrictionForce;
+
+			/*Consequently the rotation speed of the boat
```
Comment (superboum) [inline TarponSimulator2017/Core/Physics.cs]: Thanks for putting comments ;)

Generally, when you use multi lines comments, you would use them like it:

```csharp
/*
 * Consequently the rotation speed of the boat
 * depends of its speed
 */
```

File: TarponSimulator2017/Core/Physics.cs [hunk 5]
```diff
@@ -1,11 +1,137 @@
...
+
+		public void Turn(Utils.Direction d)
+		{
+			TurnAngleSpeed = (int)d * TurnSpeed;
+			Console.WriteLine ("Turn triggered");
+		}
+
+		public void ComputeMovement(int now)
+		{
+			Console.WriteLine ("ComputeMovement triggered");
```
Comment (superboum) [inline TarponSimulator2017/Core/Physics.cs]: Do we really need to keep this debug line ?

File: TarponSimulator2017/Core/Physics.cs [hunk 6]
```diff
@@ -1,11 +1,137 @@
...
+		public void Turn(Utils.Direction d)
+		{
+			TurnAngleSpeed = (int)d * TurnSpeed;
+			Console.WriteLine ("Turn triggered");
+		}
+
+		public void ComputeMovement(int now)
+		{
+			Console.WriteLine ("ComputeMovement triggered");	
+			elapsedTime = now;
```
Comment (superboum) [inline TarponSimulator2017/Core/Physics.cs]: I know that I'm the only responsible for this, but we can really get rid of it. Just replace:

```csharp
public void ComputeMovement(int now)
```

by

```csharp
public void ComputeMovement(int elapsedTime)
```

File: TarponSimulator2017/TarponGame.cs [hunk 7]
```diff
@@ -51,21 +51,15 @@ protected override void Initialize()
 
 			World world = new World (); 
 
-			scene = SceneInGame.Instance;
-		
+			//Deso Lucie, j'ai mis un bye a ton singleton, je sais pas comment ça marche ptdr
```
Comment (superboum) [inline TarponSimulator2017/TarponGame.cs]: Nice x)

File: TarponSimulator2017/Core/Boat.cs (hunk 2 above)
Comment (Hamilcare) [reply TarponSimulator2017/Core/Boat.cs]: Let's go for esperanto

**Pull request merged**